### PR TITLE
Feature:"Adding method to get the product qty from qty of ingredients in the i…

### PR DESCRIPTION
**Pull request by eapg**
*Adding method to get the quantity of products that can be made from ingredients in the inventory.

This method from a product_ids list will return a map of product id keys and quantity of products that can be made value.

the method will evaluate the ingredient need to made a product will the lowest quantity on the inventory to determine the quantity of products that can be made.

example:

if we want to know how many hamburger can be made from ingredients in the inventory:

hamburguer ingredients:

qty    ingredient
2        cheese
2        bread
1        meat
2       tomatoes
2      bacon

inventory ingredients:

qty      ingredient
100       cheese
100       bread
50         meat
100       tomato
20         bacon

if we dived the ingredients in the invetory divided by product ingredient we get the next result:

cheese 100/2 = 50
bread   100/2 = 50
meat     50/2  = 25
tomato 100/2 = 50
bacon    20/2 = 10

as bacon is the lowes ingredient in the inventory it means that the quantity of hamburger that can be made are 10.

### CODE REVIEW

File: src/lib/repositories/impl/inventory_ingredient_repository_impl.py [hunk 1]
```diff
@@ -69,3 +73,26 @@ def validate_ingredient_availability(
...
+            quantity_ingredients_by_product = reduce(
+                lambda quantity_ingredients_result, product_ingredient: products_qty_by_ingredients_qty_reducer(
+                    quantity_ingredients_result,
+                    product_ingredient,
+                    self.get_by_ingredient_id(product_ingredient.ingredient_id),
+                ),
+                product_ingredients,
+                [],
+            )
+            quantity_product_that_can_be_made_map[product_id] = min(
```
Comment (reynoldmorel) [inline src/lib/repositories/impl/inventory_ingredient_repository_impl.py:95]: Why do you need min? I don't think this is  needed, can you elaborate?

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/lib/repositories/impl/inventory_ingredient_repository_impl.py [hunk 2]
```diff
@@ -69,3 +73,26 @@ def validate_ingredient_availability(
         if ingredient_to_validate[0].ingredient_quantity > quantity_to_use:
             return True
         return False
+
+    def get_final_product_qty_by_product_ids(self, product_ids):
+
+        quantity_product_that_can_be_made_map = {}
```
[the remark this answers is not in the document]
Comment (eapg) [reply src/lib/repositories/impl/inventory_ingredient_repository_impl.py:79]: done.

File: src/lib/repositories/impl/inventory_ingredient_repository_impl.py [hunk 3]
```diff
@@ -69,3 +74,25 @@ def validate_ingredient_availability(
         if ingredient_to_validate[0].ingredient_quantity > quantity_to_use:
             return True
         return False
+
+    def get_final_product_qty_by_product_ids(self, product_ids):
+
+        final_product_possible_qty_map = reduce(
```
Comment (reynoldmorel) [inline src/lib/repositories/impl/inventory_ingredient_repository_impl.py:80]: Try to do something like:
```
reducer1 = reduce(...)
reducer2 = reduce(..., reducer1)
```

This is just an idea. Just avoid nested code.

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/lib/repositories/impl/inventory_ingredient_repository_impl.py (hunk 3 above)
Comment (eapg) [reply src/lib/repositories/impl/inventory_ingredient_repository_impl.py]: done

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**